org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/ContentModelCodeLensParticipant.java

```diff
@@ -62,9 +67,60 @@
  */
 public class ContentModelCodeLensParticipant implements ICodeLensParticipant {
 
+	private final ContentModelManager contentModelManager;
+
+	public ContentModelCodeLensParticipant(ContentModelManager contentModelManager) {
+		this.contentModelManager = contentModelManager;
+	}
+
 	@Override
 	public void doCodeLens(ICodeLensRequest request, List<CodeLens> lenses, CancelChecker cancelChecker) {
-		if (!canSupport(request)) {
+		// List of referenced schema/grammar
+		createReferencedGrammarLenses(request, lenses);
+		// "Bind to grammar/schema..."
+		createBindToGrammarSchemaLenses(request, lenses);
+	}
+
+	private void createReferencedGrammarLenses(ICodeLensRequest request, List<CodeLens> lenses) {
+		if (!canSupportOpenUri(request)) {

```

Comment:
It might be interesting to create a CodeLens with the path to the schema that does nothing when clicked if `OpenUri` is not supported. What do you think?

Reply:
Yes you are right (ex : on WWD from Eclipse IDE). supported now.